### PR TITLE
Turn InheritDocstrings into an empty class

### PR DESCRIPTION
**Pull request by pllim**
As suggested by @astrofrog in https://github.com/astropy/astropy/pull/8881#issuecomment-513838778 .

### CODE REVIEW

File: astropy/utils/misc.py [hunk 1]
```diff
@@ -498,48 +498,9 @@ class InheritDocstrings(type):
...
-        ...         def wiggle(self):
-        ...             "Wiggle the thingamajig"
-        ...             pass
-        ...     class B(A):
-        ...         def wiggle(self):
-        ...             pass
-        >>> B.wiggle.__doc__
-        u'Wiggle the thingamajig'
-    """
+    .. warning:: This class is now a no-op; please use Sphinx>=1.7.
```
Comment (astrofrog) [inline astropy/utils/misc.py]: Can you spell this out more - no-op is quite jargony

Comment (pllim) [reply astropy/utils/misc.py]: Can you please provide suggested text? I don't know how to explain this other than "no-op". :grimacing:

**Comment by mhvk** [comment]
Are we sure we want this? I think we'd at least do a sphinx version test. But what is the problem being solved? I.e., why not just keep it as is? Changing what a decorator does while it is being deprecated seems rather strange.

**Comment by pllim** [comment]
I am neutral on this; simply following up on suggestion from @astrofrog , so maybe he can elaborate.

**Comment by astrofrog** [comment]
My main concern was that this code would continue to live in the LTS version which means we then need to continue supporting it for two years, whereas an empty class is easier to support :) But no strong feelings at the end of the day, so happy to go either way.

**Comment by pllim** [comment]
> this code would continue to live in the LTS version

This is the same problem with all the other stuff we deprecate in 4.0 though. 🤔

**Comment by mhvk** [comment]
Agreed with @pllim - that is no different for other deprecations. Let's just do it the regular way, we cannot know whether anyone actually relies on `__doc__` being set.

**Comment by bsipocz** [comment]
> My main concern was that this code would continue to live in the LTS version which means we then need to continue supporting it for two years,

We can always say that we don't actively support deprecated functionality but rather let it rot there waiting to be removed.

**Comment by bsipocz** [comment]
Let's close this for now. Do reopen if @astrofrog disagrees.

**Pull request closed without merging**